File: Rust/6 Kyu/simple_nearest_prime.rs
pub fn is_prime(x: u64) -> bool {
    if x == 2 {return true};
    if x < 2 || x % 2 == 0 {return false};
    (3..).step_by(2).take_while(|i| i*i <= x).all(|i| x % i != 0)
}

fn solve(n: u64) -> u64 {
    if is_prime(n){
        return n;
    }
    let mut prox = n+1;
    
    let mut ant = n-1;
    
    loop{
        if is_prime(ant){
            return ant;
        }
        if is_prime(prox){
            return prox;
        }
        prox+=1;
        ant-=1;
        
    
    }
    

File: CPP/6 Kyu/simple_nearest_prime.cpp
#include <cmath>
bool isPrime(long long x) {
  if (x==2 || x == 3){
      return true;
    }
    if (x<2 || x%2==0){
      return false;
    }
    int lim = round(sqrt(x))+1;
    for (long long i = 3; i <= lim;i+=2){
      if(x%i==0){
        return false;
      }
    }
    return true;
}



long long solve(long long n){
    if(isPrime(n)){
      return n;
}
    long long prox = n+1;
    long long ant = n-1;
    while(1){
      if(isPrime(ant)){
            return ant;
      }     
      if(isPrime(prox)){
          return prox;
      }
      
      ant-=1;
      prox+=1;
    
    }
}